### PR TITLE
session get_state should return bytes

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/14707

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_sessions_async.py [hunk 1]
```diff
@@ -741,7 +741,8 @@ async def test_async_session_get_set_state_with_receiver(self, servicebus_namesp
                 async for m in session:
                     assert m.session_id == session_id
                     count += 1
-                await session.session.get_state()
+                state = await session.session.get_state()
+                assert type(state) is bytes
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/async_tests/test_sessions_async.py:745]: can we do it
```python
assert state == b'first_state'
```
which not only checks the type but also checks the content

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py [hunk 2]
```diff
@@ -121,8 +121,6 @@ def get_state(self, **kwargs):
             timeout=timeout
         )
         session_state = response.get(MGMT_RESPONSE_SESSION_STATE)  # type: ignore
-        if isinstance(session_state, six.binary_type):
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py:124]: `aio.SerivceBusSession` should move this part as well

**Comment by yunhaoling** [comment]
thanks for the PR! one more thing is that we would need to mention it in our changelog as a breaking change.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py (hunk 2 above)
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py:124]: It won't let me comment above where relevant but the type signature needs to change too.

I'd also call out that there's an async version of all this as well that'd need to change similarly.

**Comment by yunhaoling** [comment]
LGTM, there's only changelog conflict, I will approve after after resolving and I don't think we need to re-run the tests.

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**